src/sentry/static/sentry/app/views/organizationStream/overview.jsx

```diff
@@ -415,8 +413,17 @@ const OrganizationStream = createReactClass({
     this.transitionTo({sort});
   },
 
-  onCursorChange(cursor, path, query) {
-    this.transitionTo({cursor});
+  onCursorChange(cursor, path, query, pageDiff) {
+    const queryPageInt = parseInt(query.page, 10);
+    let nextPage = isNaN(queryPageInt) ? pageDiff : queryPageInt + pageDiff;
```

Comment:
In the `-1` case should `nextPage` be 0?

Reply:
Oops, fixed so that if nextPage is <= 0, we reset cursor/page.